compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -930,6 +930,7 @@ impl<'a> Parser<'a> {
     /// TRAIT_BOUND_MODIFIERS = [["~"] "const"] ["async"] ["?" | "!"]
     /// ```
```

Comment:
````suggestion
    /// CONSTNESS = [["~"] "const"]
    /// ASYNCNESS = ["async"]
    /// POLARITY = ["?" | "!"]
    /// ```
    ///
    /// See [`Self::parse_generic_ty_bound`] for the complete grammar of trait bound modifiers.
````

Reply:
Yuup, the discrepancy between the grammar snippets & the recursive descent parser (not semantically only organizationally) is giving me the ick

Reply:
Also, the EBNF snippets in `rustc_parse` are like my worst nightmare lol, switching between different styles (EBNF is an open *family* of dialects anyway practically speaking) and being beyond outdated. I've been meaning to make everything consistent in one big PR for a while now, but er, not sure if it's worth my energy ^^'